static-metric/src/parser.rs

```diff
@@ -14,28 +14,23 @@
 use std::collections::HashMap;
 
 use proc_macro2::Span;
-use syn::buffer::Cursor;
+use syn::parse::{Parse, ParseStream};
 use syn::punctuated::Punctuated;
-use syn::synom::{PResult, Synom};
+use syn::token::*;
 use syn::*;
 
 /// Matches `label_enum` keyword.
 struct LabelEnum {
     pub span: Span,
 }
 
-impl Synom for LabelEnum {
-    fn parse(tokens: Cursor) -> PResult<Self> {
-        if let Some((term, rest)) = tokens.term() {
-            if term.as_str() == "label_enum" {
-                return Ok((LabelEnum { span: term.span() }, rest));
-            }
+impl Parse for LabelEnum {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let ident: Ident = input.parse()?;
+        if &ident.to_string() == "label_enum" {
```

Comment:
Is the `.to_string()` here an unnecessary copy?

Reply:
It is a copy, I'm not sure if it is unnecessary - I don't think there is a better way of getting a string repr of an `Ident`, and turning a string into an ident also requires a copy. It also doesn't matter - the ident name will always be short and this code only executes at compile time and is not hot.

Reply:
@nrc `syn 0.15.44` has [this impl](https://docs.rs/syn/0.15.44/syn/struct.Ident.html#impl-PartialEq%3CT%3E):

```rust
impl<T> PartialEq<T> for Ident where
    T: AsRef<str> + ?Sized, 
```

so `ident == "label_enum"` (add or minus a few `&` and `*`) should just work.